utils/docker/vcluster/daos-cm.sh

```diff
@@ -0,0 +1,230 @@
+#!/bin/bash
+
+set -e
+set -o pipefail
+
+VERSION=0.1
+CWD="$(realpath $(dirname $0))"
+
+DAOS_POOL_SIZE=10G
+
+ANSI_COLOR_BLACK=30
+ANSI_COLOR_RED=31
+ANSI_COLOR_GREEN=32
+ANSI_COLOR_YELLOW=33
+ANSI_COLOR_BLUE=34
+ANSI_COLOR_MAGENTA=35
+ANSI_COLOR_CYAN=36
+ANSI_COLOR_WHITE=37
+ANSI_COLOR_BRIGHT_BLACK=90
+ANSI_COLOR_BRIGHT_RED=91
+ANSI_COLOR_BRIGHT_GREEN=92
+ANSI_COLOR_BRIGHT_YELLOW=93
+ANSI_COLOR_BRIGHT_BLUE=94
+ANSI_COLOR_BRIGHT_MAGENTA=95
+ANSI_COLOR_BRIGHT_CYAN=96
+ANSI_COLOR_BRIGHT_WHITE=97
+
+TRACE_LEVEL_QUIET=-1
+TRACE_LEVEL_STANDARD=0
+TRACE_LEVEL_VERBOSE=1
+TRACE_LEVEL_DEBUG=2
+TRACE_LEVEL=$TRACE_LEVEL_STANDARD
+
+function debug
+{
+	if [[ $TRACE_LEVEL -ge $TRACE_LEVEL_DEBUG ]]
+	then
+		echo -e "[\e[${ANSI_COLOR_GREEN}mDEBUG  \e[00m] $@"
+	fi
+}
+
+
+function info
+{
+	if [[ $TRACE_LEVEL -ge $TRACE_LEVEL_VERBOSE ]]
+	then
+		echo -e "[\e[${ANSI_COLOR_CYAN}mINFO   \e[00m] $@"
+	fi
+}
+
+function warning
+{
+	if [[ $TRACE_LEVEL -ge $TRACE_LEVEL_STANDARD ]]
+	then
+		echo -e "[\e[${ANSI_COLOR_YELLOW}mWARNING\e[00m] $@" 1>&2
+	fi
+}
+
+function error
+{
+	if [[ $TRACE_LEVEL -ge $TRACE_LEVEL_STANDARD ]]
+	then
+		echo -e "[\e[${ANSI_COLOR_BRIGHT_RED}mERROR  \e[00m] $@" 1>&2
+	fi
+}
+
+function fatal
+{
+	if [[ $TRACE_LEVEL -ge $TRACE_LEVEL_STANDARD ]]
+	then
+		echo -e "[\e[${ANSI_COLOR_RED}mFATAL  \e[00m] $@" 1>&2
+	fi
+	exit 1
+}
+
+function check_cmds
+{
+	for cmd in $@
+	do
+		{ hash $cmd > "/dev/null" 2>&1 ; } || { fatal "$cmd command not installed" ; }
+	done
+}
+
+check_cmds docker docker-compose
+
+function usage
+{
+	cat <<- EOF
+		usage: daos-cm.sh [OPTIONS] CMD [ARGS]
+
+		Manage DAOS Virtual Cluster containers
+
+		Options:
+		   -z, --pool-size <size>   total size of DAOS pool (default "10G")
+		   -h, --help               show this help message and exit
+		   -V, --version            show version number
+		   -q, --quiet              quiet mode
+		   -v, --verbose            verbose mode
+		   -D, --debug              debug mode
+
+		Commands:
+		   start <ip address>       start DAOS virtual cluster
+		   stop                     stop DAOS virtual cluster
+		   state                    display the state of the DAOS virtual cluster containers
+	EOF
+}
+
+function run
+{
+	if [[ $TRACE_LEVEL -ge $TRACE_LEVEL_STANDARD ]]
+	then
+		"$@"
+	else
+		"$@" &> /dev/null
+	fi
+}
+
+function state
+{
+	info "State of the DAOS virtual cluster containers"
+	if ! run docker-compose ps ; then
```

Comment:
FYI - this only works from the vcluster directory.  The script itself will cd correctly however.